gobblin-service/src/main/java/org/apache/gobblin/service/modules/scheduler/GobblinServiceJobScheduler.java

```diff
@@ -725,12 +725,24 @@ public void executeImpl(JobExecutionContext context) throws JobExecutionExceptio
 
       // Obtain trigger timestamp from trigger to pass to jobProps
       Trigger trigger = context.getTrigger();
-      // THIS current event has already fired if this method is called, so it now exists in <previousFireTime>
-      long triggerTimestampMillis = asUTCEpochMillis(trigger.getPreviousFireTime());
-      jobProps.setProperty(ConfigurationKeys.SCHEDULER_EVENT_TO_TRIGGER_TIMESTAMP_MILLIS_KEY,
-          String.valueOf(triggerTimestampMillis));
-      _log.info(jobSchedulerTracePrefixBuilder(jobProps) + "triggerTime: {} nextTriggerTime: {} - Job triggered by "
-              + "scheduler", triggerTimestampMillis, asUTCEpochMillis(trigger.getNextFireTime()));
+      long triggerTimestampMillis;
+      // If the trigger is a reminder type event then utilize the trigger time saved in job properties rather than the
+      // actual firing time
+      if (jobDetail.getKey().getName().contains("reminder")) {
+        triggerTimestampMillis = Long.parseLong(
+            jobProps.getProperty(ConfigurationKeys.SCHEDULER_EVENT_TO_TRIGGER_TIMESTAMP_MILLIS_KEY, "0"));
+        long eventToRevisit = Long.parseLong(
+            jobProps.getProperty(ConfigurationKeys.SCHEDULER_EVENT_TO_REVISIT_TIMESTAMP_MILLIS_KEY, "0"));
```

Comment:
I'm getting these confused, but want to ensure they're not mixed up... where could I find a comment explaining the difference (or is it worth adding one here)?

Reply:
I'll put a comment in `ConfigurationKeys` where they are defined because they are also referenced in `FlowTriggerHandler` and it's confusing there too I know.